### PR TITLE
refactor(core): Refactor proof of work tasks for unlock wallet

### PR DESCRIPTION
**Pull request by amanCypherock**
https://app.clickup.com/t/9002019994/PRF-4859

Implements the required refactor for proof of work task handling.

### CODE REVIEW

**Comment by rohanagarwal94** [comment]
Task linked: [PRF-4859 Wallet Unlock operation](https://app.clickup.com/t/9002019994/PRF-4859)

File: common/libraries/proof_of_work/pow.c [hunk 1]
```diff
@@ -176,42 +221,45 @@ void start_proof_of_work_task(const char *name) {
...
+bool proof_of_work_task() {
+  if (!pow_started) {
+    return false;
+  }
+
+  bool result = false;
+
+  lv_task_handler();
+  // The counter will run for x secs if limit = SECS_TO_HASHES(x)
+  uint16_t limit = SECS_TO_HASHES(100);
```
[the remark this answers is not in the document]
Comment (amanCypherock) [reply common/libraries/proof_of_work/pow.c]: This wasn't supposed to be commited but missed it due to overflow of the limit variable.

[the remark this answers is not in the document]
Comment (amanCypherock) [reply common/libraries/proof_of_work/pow.c]: reverted

File: common/libraries/proof_of_work/pow.c [hunk 2]
```diff
@@ -176,42 +221,45 @@ void start_proof_of_work_task(const char *name) {
...
-
-      increament_byte_array(nonce, POW_NONCE_SIZE);
+bool proof_of_work_task() {
+  if (!pow_started) {
+    return false;
+  }
+
+  bool result = false;
+
+  lv_task_handler();
```
[the remark this answers is not in the document]
Comment (amanCypherock) [reply common/libraries/proof_of_work/pow.c:247]: added

**Review by ujjwal-cyph — APPROVED** [review]

**Pull request merged**